src/coreclr/jit/compiler.cpp

```diff
@@ -2275,64 +2275,51 @@ void Compiler::compSetProcessor()
     {
         instructionSetFlags.AddInstructionSet(InstructionSet_Vector128);
     }
+
     if (instructionSetFlags.HasInstructionSet(InstructionSet_AVX))
     {
         instructionSetFlags.AddInstructionSet(InstructionSet_Vector256);
     }
-    // x86-64-v4 feature level supports AVX512F, AVX512BW, AVX512CD, AVX512DQ, AVX512VL and
-    // AVX512F/AVX512BW/AVX512CD/AVX512DQ/VX512VL have been shipped together historically.
-    // It is therefore unlikely that future CPUs only support "just one" and
-    // not worth the additional complexity in the JIT to support.
-    if (instructionSetFlags.HasInstructionSet(InstructionSet_AVX512F) &&
-        instructionSetFlags.HasInstructionSet(InstructionSet_AVX512BW) &&
-        instructionSetFlags.HasInstructionSet(InstructionSet_AVX512DQ))
-    {
-        // Using JitStressEVEXEncoding flag will force instructions which would
-        // otherwise use VEX encoding but can be EVEX encoded to use EVEX encoding
-        // This requires AVX512VL support. JitForceEVEXEncoding forces this encoding, thus
-        // causing failure if not running on compatible hardware.
 
-        // We can't use !DoJitStressEvexEncoding() yet because opts.compSupportsISA hasn't
-        // been set yet as that's what we're trying to set here
+    // x86-64-v4 feature level supports AVX512F, AVX512BW, AVX512CD, AVX512DQ, AVX512VL
+    // These have been shipped together historically and at the time of this writing
+    // there exists no hardware which doesn't support the entire feature set. To simplify
+    // the overall JIT implementation, we currently require the entire set of ISAs to be
+    // supported and disable AVX512 support otherwise.
```

Comment:
Like the comment says, we want to support AVX512 only if all of `AVX512F`, `AVX512BW`, `AVX512CD`, `AVX512DQ`, and `AVX512VL` are supported.

These ISAs form the `x86-64-v4` baseline and there has never shipped a piece of hardware without all of them.

Some examples of things we'd have to consider are that legacy-encoded `xorps` is `SSE` and VEX-encoded `vxorps` is AVX. However, EVEX-encoded `xorps` is AVX512DQ. Likewise the EVEX support for XMM/YMM based `xorps` is then `AVX512DQ + AVX512VL` (`AVX512DQ_VL`).

Supporting this "properly" would require us to add some fairly complex checks to import and likely LSRA to handle the difference and ensure that some suitable fallback is generated. We could write all the logic to support them, but without hardware existing that will will be "needless" overhead and negatively impact throughput. So its much easier to just write the JIT to disable AVX512 entirely if any of the "core" ISAs are unsupported.